### ISSUE TITLE
Panoramic night vision goggles sometimes can't be placed in army helmets

### ISSUE THREAD

**Issue by ISuckM8** [CleverRaven/Cataclysm-DDA#76707]
### Describe the bug

Depending on whether the army helmet is small or large, you won't be able to put the goggles in them. I don't know whether it's intended or not, but this is quite janky.

### Attach save file

N/A

### Steps to reproduce

1. Spawn a lot of differently-sized army helmets
2. Try attaching the goggles to each one

### Expected behavior

While I've never used them in real life and can't find information about this, I doubt this is something that would realistically happen or at least the army might provide a variant for smaller helmets.

### Versions and configuration

- OS: Linux
    - OS Version: LSB Version: n/a; Distributor ID: EndeavourOS; Description: EndeavourOS Linux; Release: rolling; Codename: rolling; 
- Game Version: cdda-experimental-2024-09-20-0455 4254b22 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth]
]

### Additional context

If this is not intended, I'll fix it myself.

**Comment by tcld** [comment]
I've experiences this as well. Despite the pocket sizes of the helmets being sufficient, I was unable to attach a set of panoramic (**the same goes for the enhanced version**) night vision goggles to any of the helmets I've found/tested. Workaround was to use the "pocket manager" (pressing i-i) and placing them manually. (So, yes, definitely janky.)

**Comment by LNSmith458** [comment]
Additional info: It doesn't appear to be limited to Panoramic NVGs.
A handy screenshot: 
![image](https://github.com/user-attachments/assets/9987e91f-bab7-4fd5-98a3-8d87afbec78d)

A handy link:
https://github.com/CleverRaven/Cataclysm-DDA/blob/2e5b3cfe9ad582fb031dc7b1b5a2f11f84e45f9a/src/item_pocket.cpp#L1559-L1564

**Comment by AlexMooney** [comment]
It seems like there's a workaround where you can insert NVG into a helmet if you're not wearing it.

**Comment by tenehea** [comment]
I encountered a situation where simply not wearing it actually wasn't enough. I had to completely strip down (which ALSO wasn't enough) AND don the helmet on its own and then it let me insert the panoramic night vision.

**Comment by AnomalyXIII** [comment]
yep, once i read Littles, little entry there it was obvvious and i took off my safty glasses and bam went right in. thanks.

**Comment by marilynias** [comment]
Are we saying that NVG's should be wearable with glasses/goggles? If yes the solution would be adding 
```json
"layers": [ "SKINTIGHT", "NORMAL" ],
"rigid_layer_only": true,
```
to all glasses/goggles as done in #79621.
that might make this a "good first issue" as its a simple, but repetitive fix

**Comment by Shados** [comment]
Helmet attachments are implemented using the "ablative armor" pockets mechanic, and I am fairly certain this issue is related to the fact that the `rigid_layer_only` flag doesn't work with things placed in ablative armor pockets.

The code for handling ablative armor pockets doesn't use the `outfit::check_rigid_conflicts()` function and instead re-implements the rigidity logic in order to work-around pockets not tracking the item they're on (and items in turn not tracking the character wearing them, if any), and currently it just... doesn't take into account the `rigid_layer_only` flag at all.
